lib/Core/MakefileDepsParser.cpp

```diff
@@ -70,6 +70,12 @@ static void skipNonNewlineWhitespace(const char*& cur, const char* end) {
       continue;
     }
 
+    // Also skip \r\n newlines
+    if (c == '\\' && cur + 2 != end && cur[1] == '\r' && cur[2] == '\n') {
```

Comment:
I think that needs to be something like `cur + 2 < end`.  If we have only 1 byte left, the +2 would miss that we're going to read past the end.

Reply:
One way to show that would be to add a test for `input = "out: \\\r";`

Reply:
Good point, fixed.